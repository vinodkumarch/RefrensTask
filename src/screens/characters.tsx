import * as React from 'react';
import {ActivityIndicator, FlatList} from 'react-native';
import {CharacterItem} from '../components/characterItem';
import {CharacterStyles} from './characters.styles';
import {useCharacterList} from './useCharacterList';

export const CharactersList = () => {
  const {charactersData, loading, onCharacterProfilePress} = useCharacterList();
  return loading ? (
    <ActivityIndicator
      size={'large'}
      style={CharacterStyles.loadingIndicator}
    />
  ) : (
    <FlatList
      testID="characters.mainContainer"
      style={CharacterStyles.container}
      data={charactersData}
      renderItem={({item, index}) => (
        <CharacterItem
          character={item}
          onCharacterProfilePress={() => onCharacterProfilePress(index)}
        />
      )}
      numColumns={2}
    />
  );
};
