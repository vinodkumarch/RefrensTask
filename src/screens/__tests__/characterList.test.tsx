import React from 'react';
import {CharactersList} from '../characters';
import {render, waitFor} from '@testing-library/react-native';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

describe('UT for Character screen', () => {
  it('should reder the Character List', async () => {
    const {toJSON, getByTestId} = render(<CharactersList />);
    await waitFor(() =>
      expect(getByTestId('characters.mainContainer')).toBeTruthy(),
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
