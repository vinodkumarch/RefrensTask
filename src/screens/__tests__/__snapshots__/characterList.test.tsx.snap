// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UT for Character screen should reder the Character List 1`] = `
<RCTScrollView
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flexGrow": 1,
      "marginTop": 20,
    }
  }
  testID="characters.mainContainer"
  viewabilityConfigCallbackPairs={Array []}
>
  <View />
</RCTScrollView>
`;
